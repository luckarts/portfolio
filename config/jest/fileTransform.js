const path = require('path');

// This is a custom Jest transformer turning file imports into filenames.
// https://jest-bot.github.io/jest/docs/webpack.html

module.exports = {
  process(src, filename) {
    return `module.exports = ${JSON.stringify(path.basename(filename))};`;
  }
};