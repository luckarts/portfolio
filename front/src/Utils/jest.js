import checkPropTypes from "check-prop-types";
import { createStore } from "redux";
import rootReducer from "store/reducer";

export const checkProps = (component, expectedProps) => {
  // @ts-ignore
  const ErrProps = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );

  return ErrProps;
};
export const testStore = initialState => {
  return createStore(rootReducer, initialState);
};
