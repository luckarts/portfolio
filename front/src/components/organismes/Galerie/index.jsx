import React, { useState, useEffect } from 'react';
import LightboxGalerie from '../Lightbox';

import Card from 'components/molecules/Card';
import 'react-image-lightbox/style.css';
import './style.scss';

import { Masonry } from 'lucky-react-component';
import axios from 'axios';

const Galerie = ({ photoIndex, getIndexPhoto, ...props }) => {
	const [lightboxIsOpen, setIsOpen] = useState(false);
	const [photos, setGallery] = useState([false]);
	useEffect(() => {
		async function fetchData() {
			const res = await axios('/api/gallery/get/gallery');
			setGallery(res.data.gallery);
		}
		fetchData();
	}, []);
	const openLightbox = (index) => {
		getIndexPhoto(index);
		setIsOpen(true);
	};

	const closeLightbox = (bool) => {
		setIsOpen(bool);
	};

	return (
		<div className="full-container masonry">
			<Masonry columnsCountBreakPoints={{ 350: 2, 750: 2, 900: 4 }}>
				{photos.map((photo, i) => (
					<div key={i} onClick={() => openLightbox(i)}>
						<Card i={i} className="card card-masonry" photo={photo.image} />
					</div>
				))}
			</Masonry>
			<LightboxGalerie
				photoIndex={photoIndex}
				lightboxIsOpen={lightboxIsOpen}
				closeLightbox={closeLightbox}
				getIndexPhoto={getIndexPhoto}
				photos={photos}
			/>
		</div>
	);
};

export default Galerie;
