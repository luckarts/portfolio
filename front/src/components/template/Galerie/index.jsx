import React from 'react';
import PropTypes from 'prop-types';
import Masonry from '../../organismes/Galerie';
import { connect } from 'react-redux';
import Navbar from '../../organismes/Navbar';
import { getIndexPhoto } from 'store/galeriePhoto/actions';
import Link from '../../atoms/Button/link';
import { Helmet } from 'react-helmet';

import '../style.scss';
const propTypes = {
	photoIndex: PropTypes.number,
	getIndexPhoto: PropTypes.func
};

const Galerie = ({ edit, getIndexPhoto, photoIndex, isAuthenticated }) => {
	return (
		<div className="home-page">
			<Helmet>
				<title>Artworks</title>
				<meta property="og:title" content="Artworks"></meta>
				<meta property="og:url" content="https://bachelart.fr/illustrations"></meta>
				<meta name="description" content="Artworks / Illustrations / 2D / Photoshop " />
			</Helmet>
			<Navbar />
			<div className="main-content">
				<div className="section-title-header">
					<h1 className="primary after">Mes illustrations</h1>
				</div>
				{isAuthenticated && <Link addIcon title="Importer une illustration" href="/add/illustration" />}

				<Masonry getIndexPhoto={getIndexPhoto} photoIndex={photoIndex} />
			</div>
		</div>
	);
};
export const mapStateToProps = state => ({
	photoIndex: state.GetIndex.photoIndex,
	isAuthenticated: state.user.user
});
Galerie.propTypes = propTypes;

export default connect(mapStateToProps, { getIndexPhoto })(Galerie);
