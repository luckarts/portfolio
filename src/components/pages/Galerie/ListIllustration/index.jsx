import React, { useState } from 'react';
import { Loader, Masonry, LazyImage, Lightbox } from 'components/shared';
import fetchData from 'utils/fetchData';
import api from 'API/api';
import 'react-image-lightbox/style.css';
import './style.scss';

const ListIllustration = ({ photoIndex, getIndexPhoto, ...props }) => {
  const [lightboxIsOpen, setIsOpen] = useState(false);

  const { loading, error, data } = fetchData(api.artworks.getArtworks);

  const openLightbox = (index) => {
    getIndexPhoto(index);
    setIsOpen(true);
  };

  const closeLightbox = (bool) => {
    setIsOpen(bool);
  };

  return loading ? (
    <div className="my-24 text-center ">
      <Loader bg="bg-primary " />
    </div>
  ) : error ? (
    <span data-testid="error" className="text-red-800 text-center w-full">
      {error}
    </span>
  ) : (
    <div className="w-11/12 mt-12 mb-6 m-auto masonry">
      <Masonry columnsCountBreakPoints={{ 350: 2, 750: 2, 900: 4 }}>
        {data &&
          data.gallery.map((photo, i) => (
            <div key={i} onClick={() => openLightbox(i)}>
              <div className="cursor-pointer ">
                {photo && (
                  <LazyImage
                    className="rounded-lg w-full mt-4 object-fill transition duration-300 ease-in-out hover:scale-105"
                    key={i}
                    src={'/img/' + photo.image}
                    alt={photo.image}
                    placeholder={'/thumb/img/' + photo.image}
                  />
                )}
              </div>
            </div>
          ))}
      </Masonry>
      <Lightbox
        photoIndex={photoIndex}
        lightboxIsOpen={lightboxIsOpen}
        closeLightbox={closeLightbox}
        getIndexPhoto={getIndexPhoto}
        photos={data.gallery}
      />
    </div>
  );
};

export default ListIllustration;
