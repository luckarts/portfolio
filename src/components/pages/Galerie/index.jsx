import React from 'react';
import PropTypes from 'prop-types';
import Masonry from './ListIllustration';
import { connect } from 'react-redux';
import { getIndexPhoto } from 'store/galeriePhoto/actions';
import { Helmet } from 'react-helmet';
import { Typography } from 'components/shared';

const propTypes = {
  photoIndex: PropTypes.number,
  getIndexPhoto: PropTypes.func
};

const Galerie = ({ edit, getIndexPhoto, photoIndex, isAuthenticated }) => {
  return (
    <>
      <Helmet>
        <title>Artworks</title>
        <meta property="og:title" content="Artworks"></meta>
        <meta property="og:url" content="https://bachelart.fr/illustrations"></meta>
        <meta name="description" content="Artworks / Illustrations / 2D / Photoshop " />
      </Helmet>

      <Typography variante="h2" primary className=" text-center  afterTitle">
        Mes illustrations
      </Typography>

      {isAuthenticated && (
        <Typography variante="link" addIcon href="/add/illustration">
          Importer une illustration
        </Typography>
      )}

      <Masonry getIndexPhoto={getIndexPhoto} photoIndex={photoIndex} />
    </>
  );
};
export const mapStateToProps = (state) => ({
  photoIndex: state.galeriePhoto.photoIndex,
  isAuthenticated: state.user.user
});
Galerie.propTypes = propTypes;

export default connect(mapStateToProps, { getIndexPhoto })(Galerie);
