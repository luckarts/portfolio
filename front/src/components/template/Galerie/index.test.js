import React from "react";
import { shallow, mount } from "enzyme";
import Galerie, { mapStateToProps } from ".";
import { checkProps, testStore, findByTestAtrr } from "Utils/jest";

let wrapper, initialState, props, propTypes;
props = {
  getIndexPhoto: jest.fn(),
  photoIndex: 0
};

const setUp = (initialState = {}) => {
  const store = testStore(initialState);
  const wrapper = shallow(<Galerie store={store} {...props} />)
    .childAt(0)
    .dive();

  return wrapper;
};
describe("BannerHeader Component", () => {
  beforeEach(() => {
    (initialState = {
      GetIndex: {
        photoIndex: 0
      }
    }),
      (wrapper = setUp(initialState));
  });
  it("Should renderss.", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
describe("maps", () => {
  it("mapStateToProps", () => {
    expect(mapStateToProps(initialState)).toEqual(initialState.GetIndex);
  });
});
