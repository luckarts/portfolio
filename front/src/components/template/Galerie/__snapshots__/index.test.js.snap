// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerHeader Component Should renderss. 1`] = `ShallowWrapper {}`;
